### PR TITLE
Add CAASApplicationProvisioner facade.

### PR DESCRIPTION
**Pull request by hpidcock**
## Description of change

Adds CAASApplicationProvisioner facade to provide charm, provisioning info, password setting and unit garbage collection.

The purpose behind API and APIGroup for the apiserver facades is to keep testing of the API and it's required state to just the new facade and not the common facade code.

## QA steps

Unit Tests. Need worker to test thoroughly.

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: cloudconfig/podcfg/image.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 const (
 	JujudOCINamespace = "jujusolutions"
 	JujudOCIName      = "jujud-operator"
+	JujuK8sAgentName  = "k8sagent"
```
Comment (ycliuhw) [inline cloudconfig/podcfg/image.go:18]: have we already split `k8sagent` from `jujud-operator`(the build process)?

File: cloudconfig/podcfg/image.go [hunk 2]
```diff
@@ -51,6 +52,18 @@ func GetJujuOCIImagePath(controllerCfg controller.Config, ver version.Number, bu
 	return imageRepoToPath(controllerCfg.CAASImageRepo(), ver)
 }
 
+// GetJujuK8sOCIImagePath returns the jujud oci image path.
```
Comment (ycliuhw) [inline cloudconfig/podcfg/image.go:55]: comment needs to be changed

**Review by ycliuhw — APPROVED** [review]
lgtm, thanks!

File: cloudconfig/podcfg/image.go (hunk 1 above)
Comment (hpidcock) [reply cloudconfig/podcfg/image.go:18]: Yep that was landed earlier

**Pull request merged**